modules/utxo-lib/src/bitgo/UtxoPsbt.ts

```diff
@@ -282,7 +282,12 @@ export class UtxoPsbt<Tx extends UtxoTransaction<bigint> = UtxoTransaction<bigin
   finalizeAllInputs(): this {
     checkForInput(this.data.inputs, 0); // making sure we have at least one
     this.data.inputs.map((input, idx) => {
-      return input.tapScriptSig?.length ? this.finalizeTaprootInput(idx) : this.finalizeInput(idx);
+      try {
+        return input.tapScriptSig?.length ? this.finalizeTaprootInput(idx) : this.finalizeInput(idx);
+      } catch (e) {
+        // Likely a single leaf script and single sig reveal txn, try finalizing with the respective method
```

Comment:
not sure why you assume that is likely - there can be any number of reasons why finalization fails

can you think of some other way to detect the `SingleLeafScriptAndSignature` condition?